### PR TITLE
fix(hatchery:vsphere): update vmware deps + fix auth secrets

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/hatchery/vsphere/spawn.go [hunk 1]
```diff
@@ -39,7 +39,11 @@ func (h *HatcheryVSphere) SpawnWorker(ctx context.Context, spawnArgs hatchery.Sp
 
 	if errM != nil || spawnArgs.Model.NeedRegistration {
 		// Generate worker model vm
-		vm, errV = h.createVMModel(*spawnArgs.Model)
+		vm, errV = h.createVMModel(*spawnArgs.Model, spawnArgs.WorkerName)
+		if errV != nil {
+			log.Error(ctx, "Unable to create VM Model: %v", errV)
+			return errV
+		}
```
Comment (fsamin) [inline engine/hatchery/vsphere/spawn.go:46]:
```suggestion
                var err error
		vm, err = h.createVMModel(*spawnArgs.Model, spawnArgs.WorkerName)
		if err != nil {
			log.Error(ctx, "Unable to create VM Model: %v", err)
			return err
		}
```

File: engine/hatchery/vsphere/spawn.go [hunk 2]
```diff
@@ -39,7 +39,11 @@ func (h *HatcheryVSphere) SpawnWorker(ctx context.Context, spawnArgs hatchery.Sp
...
 	}
 
 	if vm == nil || errV != nil {
```
Comment (fsamin) [inline engine/hatchery/vsphere/spawn.go:49]: is the test on errBV still useful ?

File: engine/hatchery/vsphere/spawn.go [hunk 3]
```diff
@@ -58,7 +62,7 @@ func (h *HatcheryVSphere) SpawnWorker(ctx context.Context, spawnArgs hatchery.Sp
 		Created:                 time.Now(),
 	}
 
-	cloneSpec, folder, errCfg := h.createVMConfig(vm, annot)
+	cloneSpec, folder, errCfg := h.createVMConfig(vm, annot, spawnArgs.WorkerName)
```
Comment (fsamin) [inline engine/hatchery/vsphere/spawn.go:65]: s/errCfg/err

File: engine/hatchery/vsphere/spawn.go [hunk 4]
```diff
@@ -75,13 +79,24 @@ func (h *HatcheryVSphere) SpawnWorker(ctx context.Context, spawnArgs hatchery.Sp
 		return sdk.WrapError(errW, "state in error")
 	}
 
-	return h.launchScriptWorker(spawnArgs.WorkerName, spawnArgs.JobID, spawnArgs.WorkerToken, *spawnArgs.Model, spawnArgs.RegisterOnly, info.Result.(types.ManagedObjectReference))
+	// Wait for IP
+	vmWorker := object.NewVirtualMachine(h.vclient.Client, info.Result.(types.ManagedObjectReference))
+	ip, errW := vmWorker.WaitForIP(ctx, true)
```
Comment (fsamin) [inline engine/hatchery/vsphere/spawn.go:84]: s/errW/err

File: engine/hatchery/vsphere/vsphere.go [hunk 5]
```diff
@@ -111,6 +122,15 @@ func (h *HatcheryVSphere) CheckConfiguration(cfg interface{}) error {
 		return fmt.Errorf("vsphere-datacenter is mandatory")
 	}
 
+	if hconfig.IPRange != "" {
+		ips, err := sdk.IPinRanges(context.Background(), hconfig.IPRange)
+		if err != nil {
+			return fmt.Errorf("flag or environment variable openstack-ip-range error: %s", err)
```
Comment (fsamin) [inline engine/hatchery/vsphere/vsphere.go:128]:
```suggestion
			return fmt.Errorf("flag or environment variable openstack-ip-range error: %v", err)
```

File: sdk/hatchery/model.go [hunk 6]
```diff
@@ -17,9 +17,17 @@ func ModelInterpolateSecrets(hWithModels InterfaceWithModels, model *sdk.Model)
...
 	}
 
-	model.ModelDocker.Password, err = interpolate.Do(model.ModelDocker.Password, modelSecrets.ToMap())
-	if err != nil {
-		return sdk.WrapError(err, "cannot interpolate registry password for model %s", model.Path())
+	switch {
+	case model.Type == sdk.Docker && model.ModelDocker.Private:
+		model.ModelDocker.Password, err = interpolate.Do(model.ModelDocker.Password, modelSecrets.ToMap())
+		if err != nil {
+			return sdk.WrapError(err, "cannot interpolate registry password for model %s", model.Path())
```
Comment (fsamin) [inline sdk/hatchery/model.go:24]: is it still a registry password ?

**Review by fsamin — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15095.0 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15095.0 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ■

File: engine/hatchery/vsphere/types.go [hunk 7]
```diff
@@ -34,6 +36,13 @@ type HatcheryConfiguration struct {
 	// CardName vsphere-ethernet-card Name of the virtual ethernet card
 	VSphereCardName string `mapstructure:"cardName" toml:"cardName" default:"e1000" commented:"false" comment:"Name of the virtual ethernet card" json:"cardName"`
 
+	// IPRange IP Range
+	IPRange string `mapstructure:"iprange" toml:"iprange" default:"" commented:"false" comment:"Facultative. IP Range for spawned workers. \n Format: a.a.a.a/b,c.c.c.c/e \n Hatchery will use an IP from this range to create Virtual Machine (Fixed IP Attribute).\nIf not set, it will get an address from the neutron service" json:"iprange,omitempty"`
```
Comment (yesnault) [inline engine/hatchery/vsphere/types.go:40]:
```suggestion
	IPRange string `mapstructure:"iprange" toml:"iprange" default:"" commented:"false" comment:"Facultative. IP Range for spawned workers. \n Format: a.a.a.a/b,c.c.c.c/e \n Hatchery will use an IP from this range to create Virtual Machine (Fixed IP Attribute).\nIf not set, you have to set it in your worker model template" json:"iprange,omitempty"`
```

File: engine/hatchery/vsphere/types.go [hunk 8]
```diff
@@ -34,6 +36,13 @@ type HatcheryConfiguration struct {
...
+
+	Gateway string `mapstructure:"gateway" toml:"gateway" default:"" commented:"false" comment:"Facultative. Gateway IP for spawned workers." json:"gateway,omitempty"`
+
+	DNS string `mapstructure:"dns" toml:"dns" default:"" commented:"false" comment:"Facultative. DNS IP" json:"dns,omitempty"`
```
Comment (yesnault) [inline engine/hatchery/vsphere/types.go:44]:
```suggestion
	DNS string `mapstructure:"dns" toml:"dns" default:"" commented:"false" comment:"Optional. DNS IP" json:"dns,omitempty"`
```

**Review by yesnault — CHANGES_REQUESTED** [review]

File: sdk/hatchery/model.go (hunk 6 above)
Comment (sguiheux) [reply sdk/hatchery/model.go:24]: in this case yes

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15109.0 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15109.0 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ■

**Review by richardlt — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15172.0 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15172.0 ■   
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ■

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15191.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Pull request merged**